frontend/lib/src/components/widgets/TextInput/TextInput.tsx

```diff
@@ -144,7 +144,20 @@ class TextInput extends React.PureComponent<Props, State> {
     // If the TextInput is *not* part of a form, we mark it dirty but don't
     // update its value in the WidgetMgr. This means that individual keypresses
     // won't trigger a script re-run.
-    this.setState({ dirty: true, value })
+    if (!isInForm(this.props.element)) {
+      this.setState({ dirty: true, value })
+      return
+    }
+
+    // If TextArea *is* part of a form, we immediately update its widgetValue
+    // on text changes. The widgetValue won't be passed to the Python
+    // script until the form is submitted, so this won't cause the script
+    // to re-run.
+    this.setState({ dirty: true, value }, () =>
+      // make sure dirty is true so that enter to submit text shows
+      // make sure we commit the widget value or this bug can occur: https://github.com/streamlit/streamlit/issues/7101
```

Comment:
I feel like this comment doesn't say anything that's not already described in the one starting on L152, so we should just remove it and include the link to the issue in that comment